code/modules/asset_cache/asset_list.dm

```diff
@@ -176,6 +217,64 @@ GLOBAL_LIST_EMPTY(asset_datums)
 
 	return out.Join("\n")
 
+/datum/asset/spritesheet/proc/read_from_cache()
+	var/replaced_css = file2text("[ASSET_CROSS_ROUND_CACHE_DIRECTORY]/spritesheet.[name].css")
+
+	var/regex/find_background_urls = regex(@"background:url\('%(.+?)%'\)", "g")
+	while (find_background_urls.Find(replaced_css))
+		var/asset_id = find_background_urls.group[1]
+		var/asset_cache_item = SSassets.transport.register_asset(asset_id, "[ASSET_CROSS_ROUND_CACHE_DIRECTORY]/spritesheet.[asset_id]")
+		var/asset_url = SSassets.transport.get_asset_url(asset_cache_item = asset_cache_item)
+		replaced_css = replacetext(replaced_css, find_background_urls.match, "background:url('[asset_url]')")
+		LAZYADD(cached_spritesheets_needed, asset_id)
+
+	var/replaced_css_filename = "data/spritesheets/spritesheet_[name].css"
+	rustg_file_write(replaced_css, replaced_css_filename)
+	SSassets.transport.register_asset("spritesheet_[name].css", replaced_css_filename)
+
+	fdel(replaced_css_filename)
+
+	return TRUE
+
+/datum/asset/spritesheet/proc/send_from_cache(client/client)
+	if (isnull(cached_spritesheets_needed))
+		stack_trace("cached_spritesheets_needed was null when sending assets from [type] from cache")
+		cached_spritesheets_needed = list()
+
+	return SSassets.transport.send_assets(client, cached_spritesheets_needed + "spritesheet_[name].css")
+
+/// Returns the URL to put in the background:url of the CSS asset
+/datum/asset/spritesheet/proc/get_background_url(asset)
+	if (generating_cache)
+		return "%[asset]%"
+	else
+		return SSassets.transport.get_asset_url(asset)
+
+/datum/asset/spritesheet/proc/write_to_cache()
+	for (var/size_id in sizes)
+		fcopy(SSassets.cache["[name]_[size_id].png"].resource, "[ASSET_CROSS_ROUND_CACHE_DIRECTORY]/spritesheet.[name]_[size_id].png")
+
+	generating_cache = TRUE
+	var/mock_css = generate_css()
+	generating_cache = FALSE
+
+	rustg_file_write(mock_css, "[ASSET_CROSS_ROUND_CACHE_DIRECTORY]/spritesheet.[name].css")
```

Comment:
The only thing I can think of is some problem with `rustg_file_write`--this is being used because `text2file` appends, and I didn't want to fdel.

Reply:
But that doesn't make sense, as `rustg_file_write` works for writing `dynamic.json` to logs.

Reply:
Creating the file beforehand and filling it on the server doesn't seem to work either.